### PR TITLE
Rename crop image

### PR DESCRIPTION
**Pull request by CSSFrancis**
### Description of the change
Renamed `Signal2D.crop_image` to `Signal2D.crop_signal2D`

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

File: upcoming_changes/3197.api.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Renamed `Signal2D.crop_image` to `Signal2D.crop_signal2D`
```
Comment (jlaehne) [inline upcoming_changes/3197.api.rst]:
```suggestion
Renamed ``Signal2D.crop_image`` to :meth:`~._signals.signal2d.Signal2D.crop_signal2D`
```

**Comment by jlaehne** [comment]
This renaming looks plausible to me, but as it breaks the API, it would be great to get a few more opinions on this.

**Comment by jlaehne** [comment]
The `code-block` example in the user guide will need to be corrected, currently it reads
`im.crop(...)`

**Comment by ericpre** [comment]
Initially wanted to rebase and update the user guide and I realised that it could simplify to `crop_signal` for `Signal1D` and `Signal2D`. What do you think?

**Comment by ericpre** [comment]
@CSSFrancis, apparently I can't request a review from you on your own pull request! 😅

File: doc/user_guide/signal1d.rst [hunk 2]
```diff
@@ -12,32 +12,32 @@ signals in the Signal1D class.
 Cropping
 --------
 
-The :py:meth:`~._signals.signal1d.Signal1D.crop_signal1D` crops the
+The :py:meth:`~._signals.signal1d.Signal1D.crop_signal` crops the
 spectral energy range *in-place*. If no parameter is passed, a user interface
```
Comment (jlaehne) [inline doc/user_guide/signal1d.rst]:
```suggestion
the signal object along the signal axis (e.g. the spectral energy range)
*in-place*. If no parameter is passed, a user interface
```

File: doc/user_guide/signal1d.rst [hunk 3]
```diff
@@ -12,32 +12,32 @@ signals in the Signal1D class.
...
 .. code-block:: python
 
-    s = hs.datasets.example_signals.EDS_TEM_Spectrum()
-    s.crop_signal1D(5, 15) # s is cropped in place
+    >>> s = hs.datasets.example_signals.EDS_TEM_Spectrum()
+    >>> s.crop_signal(5, 15) # s is cropped in place
 
 Additionally, cropping in HyperSpy can be performed using the :ref:`Signal
 indexing <signal.indexing>` syntax. For example, the following crops a spectrum
 to the 5 keV-15 keV region:
```
Comment (jlaehne) [inline doc/user_guide/signal1d.rst]:
```suggestion
indexing <signal.indexing>` syntax. For example, the following crops an EDS
spectrum to the 5 keV-15 keV region:
```

File: doc/user_guide/signal1d.rst [hunk 4]
```diff
@@ -12,32 +12,32 @@ signals in the Signal1D class.
...
 
 .. code-block:: python
 
-    s = hs.datasets.example_signals.EDS_TEM_Spectrum()
-    sc = s.isig[5.:15.] # s is not cropped, sc is a "cropped view" of s
+    >>> s = hs.datasets.example_signals.EDS_TEM_Spectrum()
```
Comment (jlaehne) [inline doc/user_guide/signal1d.rst]: Does this sample dataset move to `exspy`?

(loaded several times in this file)

**Comment by ericpre** [comment]
This conflict with #3202, will come back to it once #3202 is merged.

**Pull request merged**